File: src/08.dbpointer/17.209.go
package dbpointer

func MinSubArrayLen(target int, nums []int) int {

	n := len(nums)
	res := n + 1
	sum := 0
	j := 0

	for i := 0; i < n; i++ {
		sum += nums[i]
		for sum >= target {
			if res > i-j+1 {
				res = i - j + 1
			}
			sum -= nums[j]
			j++
		}
	}

	if res > n {
		return 0
	}

	return res
}

File: src/08.dbpointer/17.209.ts
export function minSubArrayLen(target: number, nums: number[]): number {
    /**
     * 2 3 1 2 4 3
     * 0 1 2 3 4 5
     */

    const n = nums.length;
    let res = n + 1;
    let sum = 0;
    let j = 0;

    for (let i = 0; i < n; i++) {
        sum += nums[i];
        while (sum >= target) {
            res = Math.min(res, i - j + 1);
            sum -= nums[j++];
        }
    }

    return res > n ? 0 : res;
}
